hat/app/org/hatdex/hat/api/controllers/Authentication.scala

```diff
@@ -618,10 +604,101 @@ class Authentication @Inject() (
           Future.successful(Done)
         case _ =>
           logger.error(
-            s"Failed to claim HAT with Hatters. Claim details:\n$claim\nHatters response: ${response.body}"
+            s"Failed to claim HAT with Hatters. Claim details: $claim Hatters response: ${response.body}"
           )
           Future.failed(new UnknownError("HAT claim failed"))
       }
     }
   }
-}
+
+
+
+    /**
+    * Resends the email to claim/validate the HAT
+    * This is unauthenticated request
+    */
+  def handleRequestVerification: Action[ApiValidationRequest] =
+    UserAwareAction.async(parsers.json[ApiValidationRequest]) {
+      implicit request =>
+        logger.debug("Processing resend validation request")
+
+        val email = request.body.email
+        val applicationId = request.body.applicationId
+        val isSigupToken = true
+
+        // Generic message regardless if the user is found or not.
+        val uniformResponse = resendVerificationEmail
+
+        // The email from the request and the Silhouette Env do not match
+        if (email != request.dynamicEnvironment.ownerEmail) {
+          Future.successful(uniformResponse)
+        }
+        else {
+          val enventuallTtokenToSend = for {
+            token <- tokenService.retrieve(email, isSigupToken)
+            tokenToSend <- createOrResendToken(email, token)
+          } yield tokenToSend
+
+          enventuallTtokenToSend.flatMap { tokenOpt =>
+            tokenOpt match {
+              case Some(token) => {
+                // Is Owner and is NOT EmailVerified
+                val eventuallyUsers = usersService.listUsers
+                eventuallyUsers.flatMap { users => 
+                  users.find(u => (u.roles.contains(Owner()) && !(u.roles.contains(EmailVerified)))) match {
```

Comment:
`EmailVerified` needs to be an instance instead of type. I have made similar suggestion above.